File: 2020-21/contest1/prog/intermediate/triangle.py
import os
import sys

def digit_sum_octal(num):
	s = oct(num)[2:]
	res = 0
	for ch in s:
		res += (ord(ch) - ord('0'))
	return res

while(True): # You don't have to do this on hackerrank.com!
	s = 0
	d = 0
	r = 0
	try:
		strs = [x for x in input().split()]
		s,d,r = [int(strs[0],8), int(strs[1],8), int(strs[2])]
	except EOFError as e:
		break
	res = 0
	for i in range(1, r):
		for j in range(0,i+1):
			s += d
			if i == r-1: # last row
				res += digit_sum_octal(s)
	print(res)

File: 2020-21/contest1/prog/intermediate/triangle.java
import java.io.*;
import java.util.*;

public class triangle {
	public static int digit_sum_octal(int num) {
		String str = Integer.toOctalString(num);
		int sum = 0;
		for(int i = 0; i < str.length(); ++i) {
			char ch = str.charAt(i);
			ch -= '0';
			sum += ch;
		}
		return sum;
	}

	public static void main(String[] args) {
		Scanner input = new Scanner(System.in);
		while(true) { // You don't have to do this on hackerrank.com!
			int s,d,r;
			try {
				String s1 = input.next();
				s = Integer.parseInt(s1, 8);
				String s2 = input.next();
				d = Integer.parseInt(s2, 8);
			} catch(Exception e) {
				break;
			}
			r = input.nextInt();
			input.nextLine();
			int res = 0;
			for(int i = 1; i < r; ++i) {
				for(int j = 0; j <= i; ++j) {
					s += d;
					if(i == r-1) { // last row
						res += digit_sum_octal(s);
					}
				}
			}
			System.out.println(res);
		}
	}
}
